cli_tools/gce_vm_image_import/main.go

```diff
@@ -0,0 +1,348 @@
+	fatalIfError(func() error {
+		return populateZoneIfMissing(metadataGCEHolder{})
+	})
+	fatalIfError(populateRegion)
+
+	//TODO: network, subnetwork, gcsPath (create scratch bucket including regionalization, if possible)
+}
+
+type metadataGCEHolder struct{}
+
+type metadataGCE interface {
+	OnGCE() bool
+	Zone() (string, error)
+}
+
+func (m metadataGCEHolder) OnGCE() bool {
+	return metadata.OnGCE()
+}
+
+func (m metadataGCEHolder) Zone() (string, error) {
+	return metadata.Zone()
+}
+
+func populateZoneIfMissing(mgce metadataGCE) error {
+	if *zone == "" {
+		var err error
+		var aZone = ""
+		if mgce.OnGCE() {
+			aZone, err = mgce.Zone()
+		}
+
+		if err != nil {
+			return fmt.Errorf("can't infer zone: %v", err)
+		}
+		if aZone == "" {
+			return fmt.Errorf("zone is empty")
+		}
+
+		zone = &aZone
+	}
+	return nil
+}
+
+func populateRegion() error {
+	aRegion, err := getRegion()
+	if err != nil {
+		return err
+	}
+	region = &aRegion
+	return nil
+}
+
+func getRegion() (string, error) {
+	if *zone == "" {
+		return "", fmt.Errorf("zone is empty. Can't determine region")
+	}
+	zoneStrs := strings.Split(*zone, "-")
+	if len(zoneStrs) < 2 {
+		return "", fmt.Errorf("%v is not a valid zone", *zone)
+	}
+	return strings.Join(zoneStrs[:len(zoneStrs)-1], "-"), nil
+}
+
+// Updates workflow to support image import:
+// - Labels temporary and permanent resources with appropriate labels
+// - Updates instance network interfaces to not require external IP if external IP is disabled by
+//   org policy
+func updateWorkflow(workflow *daisy.Workflow) {
+	for _, step := range workflow.Steps {
+		if step.IncludeWorkflow != nil {
+			//recurse into included workflow
+			updateWorkflow(step.IncludeWorkflow.Workflow)
+		}
+		if step.CreateInstances != nil {
+			for _, instance := range *step.CreateInstances {
+				instance.Instance.Labels = addImageImportLabels(instance.Instance.Labels)
+				if *noExternalIP {
+					configureInstanceNetworkInterfaceForNoExternalIP(instance)
+				}
+			}
+		}
+		if step.CreateDisks != nil {
+			for _, disk := range *step.CreateDisks {
+				disk.Disk.Labels = addImageImportLabels(disk.Disk.Labels)
+			}
+		}
+		if step.CreateImages != nil {
+			for _, image := range *step.CreateImages {
+				image.Image.Labels = addImageImportLabels(image.Image.Labels, getImageTypeLabelKey(image))
+			}
+		}
+	}
+}
+
+func getImageTypeLabelKey(image *daisy.Image) string {
+	imageTypeLabel := "gce-image-import"
+	if strings.Contains(image.Image.Name, "untranslated") {
+		imageTypeLabel = "gce-image-import-tmp"
+	}
+	return imageTypeLabel
+}
+
+func configureInstanceNetworkInterfaceForNoExternalIP(instance *daisy.Instance) {
+	if instance.Instance.NetworkInterfaces == nil {
+		return
+	}
+	for _, networkInterface := range instance.Instance.NetworkInterfaces {
+		networkInterface.AccessConfigs = []*compute.AccessConfig{}
+	}
+}
+
+//Extend labels with image import related labels
+func addImageImportLabels(labels map[string]string,
```

Comment:
This function doesn't actually add labels, imageImportLabels would be a better name

Reply:
It's adding image import labels to an existing label map. This map is then returned. This map is applied to resources later on in the code. 

Maybe: extendLabelMapWithImageImportLabels? Kind of long though.